libpod/container_internal_linux.go

```diff
@@ -585,11 +585,44 @@ func (c *Container) generateSpec(ctx context.Context) (*spec.Spec, error) {
 
 	// Add overlay volumes
 	for _, overlayVol := range c.config.OverlayVolumes {
+		upperDir := ""
+		workDir := ""
+		for _, o := range overlayVol.Options {
```

Comment:
Is this a cut and paste of code from a different area, if so , then you should make it into a function that returns upperDir and WorkDir if set.

Reply:
Yes moved to a function